test/test_benchmarks.py

```diff
@@ -43,7 +40,10 @@ def test_discover_benchmarks(benchmarks_fixture):
     b = benchmarks.Benchmarks.discover(conf, repo, envs, [commit_hash],
                                        regex='example')
     conf.branches = old_branches
-    assert len(b) == 36
+    if util.ON_PYPY:
+        assert len(b) == 34
+    else:
+        assert len(b) == 36
```

Comment:
@mattip and I discussed that perhaps later it might make sense to have a mechanism for discovering all benchmarks, and then skipping or mark as fail or something where it isn't feasible to run.

Currently this design is because the regex is part of the classes and in the new approach the benchmark types are dynamically populated.